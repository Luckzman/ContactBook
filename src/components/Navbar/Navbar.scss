@import '../../styles/index.scss';

.navbar{
  width: 100%;
  background-color: map-get($colors, faint-bg);
  display: flex;
  justify-content: center;
  border-bottom: 2px solid map-get($colors, nav-border);
  .container{
    width: 90%;
    display: flex;
    justify-content: space-between;
    .brand{
      color: map-get($colors, logo-color);
      text-shadow: 1px 1px 10px 10px grey;
    }
    .links{
      display: flex;
      flex-direction: row;
      justify-content: center;
      align-items: center;
      position: relative;
      .favorite-link{
        margin-right: 20px;
        color: map-get($colors, faint-link);
        cursor: pointer;
        &:hover{
          color: map-get($colors , brand-color );
          ~ .pointer{
            opacity: .6;
            bottom: 7px;
          }
        }
      }
      .pointer{
        position: absolute;
        bottom: -2px;
        left: 50px;
        width: 8px;
        height: 8px;
        border-radius: 50%;
        background-color: map-get($colors, brand-color);
        background: radial-gradient(white, map-get($colors, brand-color), white);
        box-shadow: 1px 1px 5px 2px grey;
        opacity: 0;
        transition: all 0.2s;
      }
    }
  }
}

@media screen and (max-width: 414px) {
  .navbar{
    .container{
      width: 90%;
      display: flex;
      flex-direction: column;
      justify-content: center;
      padding-bottom: 5px;
    }
  }
}