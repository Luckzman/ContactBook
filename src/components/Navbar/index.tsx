import React from 'react';
import Button from '../Button';
import './Navbar.scss';

interface Props {
  handleToggleModal: () => void;
  handleLikedContact: () => void;
  displayFavorite: boolean;
  handleAllContact: () => void;
}

const Navbar: React.FC<Props> = ({ displayFavorite, handleToggleModal, handleAllContact, handleLikedContact }) => {
  return (
    <div className="navbar">
      <div className="container">
        <h3 data-testid="brand" className="brand" onClick={handleAllContact}>
          ContactBook
        </h3>
        <div className="links">
          {displayFavorite ? (
            <p data-testid="all-contact-text" className="favorite-link extra" onClick={handleAllContact}>
              All Contacts
            </p>
          ) : (
            <p data-testid="fav-contact-link" className="favorite-link" onClick={handleLikedContact}>
              Favorite Contacts
            </p>
          )}
          <div className="pointer"></div>
          <Button data-testid="navbar-btn" onClick={handleToggleModal}>
            New Contact
          </Button>
        </div>
      </div>
    </div>
  );
};

export default Navbar;
