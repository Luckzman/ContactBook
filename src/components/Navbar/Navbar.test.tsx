import React from 'react';
import { shallow } from 'enzyme';
import Navbar from './';

it('renders without crashing', () => {
  const props = { handleToggleModal: jest.fn() };
  shallow(<Navbar {...props} />);
});
