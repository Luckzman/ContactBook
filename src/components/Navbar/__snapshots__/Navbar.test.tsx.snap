// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="navbar"
>
  <div
    className="container"
  >
    <h3
      className="brand"
    >
      ContactBook
    </h3>
    <div
      className="links"
    >
      <p
        className="favorite-link"
      >
        Favorite Contacts
      </p>
      <Button>
        New Contact
      </Button>
    </div>
  </div>
</div>
`;
